### PR TITLE
MetaKeep Lambda Creation Scripts and Sample Voting Contract

### PR DESCRIPTION
**Pull request by aditya-passbird**

### CODE REVIEW

File: lambda/voting/scripts/create.mjs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    bytecode: data["bytecode"],
+  };
+  const options = {
+    method: "POST",
+    headers: headers,
+    body: JSON.stringify(requestBody),
+  };
+  console.log("Lambda creation in process...");
+
+  const result = await fetch(url, options).catch(() => {
```
[the remark this answers is not in the document]
Comment (rohan-passbird) [reply lambda/voting/scripts/create.mjs:81]: https://dev.to/anchobies/when-that-s-not-so-fetch-error-handling-with-fetch-4cce

File: lambda/voting/test/Voting.js [hunk 2]
```diff
@@ -12,87 +11,68 @@ describe("Voting contract", function () {
     Voting = await ethers.getContractFactory("Voting");
     [owner, addr1, addr2] = await ethers.getSigners();
 
-
-    votingContract = await Voting.deploy(owner.address);
+    votingContract = await Voting.deploy(owner.address, "voting");
```
Comment (rohan-passbird) [inline lambda/voting/test/Voting.js:14]: tests are failing

```
npx hardhat test
Downloading compiler 0.8.9
Warning: SPDX license identifier not provided in source file. Before publishing, consider adding a comment containing "SPDX-License-Identifier: <SPDX-License>" to each source file. Use "SPDX-License-Identifier: UNLICENSED" for non-open-source code. Please see https://spdx.org for more information.
--> contracts/Voting.sol


Compiled 2 Solidity files successfully


  Voting contract
    1) "before all" hook in "Voting contract"


  0 passing (5ms)
  1 failing

  1) Voting contract
       "before all" hook in "Voting contract":
     TypeError: Cannot read properties of undefined (reading 'getContractFactory')
      at Context.<anonymous> (test/Voting.js:11:27)
      at processImmediate (node:internal/timers:466:21)



```

File: lambda/voting/artifacts/contracts/Voting.sol/Voting.json [hunk 3]
```diff
@@ -7,13 +7,232 @@
       "inputs": [
         {
           "internalType": "address",
-          "name": "owner",
+          "name": "lambdaOwner",
```
Comment (rohan-passbird) [inline lambda/voting/artifacts/contracts/Voting.sol/Voting.json:10]: actually remove this file and update the deploy command to run hardhat compile first

File: lambda/voting/scripts/create.mjs [hunk 4]
```diff
@@ -0,0 +1,114 @@
+// Script to deploy Smart contract using Metakeep Lambda and REST API.
```
Comment (rohan-passbird) [inline lambda/voting/scripts/create.mjs:1]: any reason we are using mjs?

File: lambda/voting/contracts/imports/MetaKeepLambda.sol [hunk 5]
```diff
@@ -0,0 +1,732 @@
+// SPDX-License-Identifier: MIT 
+pragma solidity ^0.8.0;
```
Comment (rohan-passbird) [inline lambda/voting/contracts/imports/MetaKeepLambda.sol:2]: let's also create a top level folder lambda/imports
that we will use as the master import folder

File: lambda/voting/scripts/create.mjs [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+      }
+      // if the lambda creation transaction is not Queued, logs the error and exits the program.
+    } else {
+      console.log(json);
+      console.log("Lambda creation failed");
+    }
+  });
+}
+
+main();
```
Comment (rohan-passbird) [inline lambda/voting/scripts/create.mjs:114]: throw error if API_KEY is not set.

Comment (aditya-passbird) [reply lambda/voting/scripts/create.mjs:114]: done

File: lambda/voting/package.json [hunk 7]
```diff
@@ -1,20 +1,23 @@
 {
```
[the remark this answers is not in the document]
Comment (aditya-passbird) [reply lambda/voting/package.json:1]: updated the packages needed

File: lambda/voting/test/Voting.js (hunk 2 above)
Comment (aditya-passbird) [reply lambda/voting/test/Voting.js:14]: fixed

File: lambda/voting/scripts/create.mjs [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+    Accept: "application/json",
+    "Content-Type": "application/json",
+    "x-api-key": process.env.API_KEY,
+    "Idempotency-Key": "Idempotency-Key" + Math.floor(Math.random() * 10000),
+  };
+
+  const data = JSON.parse(
+    fs.readFileSync("artifacts/contracts/Voting.sol/Voting.json")
+  );
+  const developer_address = await getDeveloperWallet();
```
[the remark this answers is not in the document]
Comment (aditya-passbird) [reply lambda/voting/scripts/create.mjs:65]: fixed

File: lambda/voting/README.md [hunk 9]
```diff
@@ -1,90 +1,64 @@
...
-for your Ethereum project. It's also a great fit for learning the basics of
-smart contract development.
+for using MetaKeep's Lambda Infrastructure.
 
 This project is intended to be used with the
-[Hardhat Beginners Tutorial](https://hardhat.org/tutorial), but you should be
+[MetaKeep Lambda Creation and Invocation Tutorial](https://docs.metakeep.xyz/reference/lambda-101), but you should be
 able to follow it by yourself by reading the README and exploring its
-`contracts`, `tests`, `scripts` and `frontend` directories.
+`contracts`, `tests`, and `scripts` directories.
```
[the remark this answers is not in the document]
Comment (aditya-passbird) [reply lambda/voting/README.md:9]: added

File: lambda/voting/artifacts/contracts/Voting.sol/Voting.json (hunk 3 above)
Comment (aditya-passbird) [reply lambda/voting/artifacts/contracts/Voting.sol/Voting.json]: done

File: lambda/voting/scripts/create.mjs [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+      let transactionStatus;
+      for (let i = 0; i < 10; i++) {
+        await sleep(5000);
+        transactionStatus = await getTransactionStatus(transactionId);
+        if (transactionStatus.status == "COMPLETED") {
+          console.log("Lambda created successfully");
+          exit(0);
+        }
+      }
```
[the remark this answers is not in the document]
Comment (aditya-passbird) [reply lambda/voting/scripts/create.mjs:111]: done

File: lambda/voting/artifacts/contracts/Voting.sol/Voting.json (hunk 3 above)
Comment (aditya-passbird) [reply lambda/voting/artifacts/contracts/Voting.sol/Voting.json]: done

File: lambda/voting/contracts/imports/MetaKeepLambda.sol (hunk 5 above)
Comment (aditya-passbird) [reply lambda/voting/contracts/imports/MetaKeepLambda.sol:2]: done

File: lambda/voting/README.md [hunk 11]
```diff
@@ -34,12 +34,31 @@ For a complete introduction to MetaKeep Lambda, refer to [this guide](https://do
 
 ## Troubleshooting
```
[the remark this answers is not in the document]
Comment (aditya-passbird) [reply lambda/voting/README.md:40]: done

**Review by rohan-passbird — APPROVED** [review]
👍

**Pull request merged**